### PR TITLE
Setup backend and add api to post users into users collection

### PR DESCRIPTION
**Pull request by ayu-ayushisood**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const http = require('http');
+const express = require("express");
+const bodyParser= require('body-parser')
+const path = require("path");
+const app = express();
+const port = process.env.PORT || "8000";
+const MongoClient = require('mongodb').MongoClient;
+
+const connectionString =  "mongodb+srv://ayushi:ayushi1234@cluster0-jvytf.mongodb.net/ayushi?retryWrites=true&w=majority"
```
Comment (karanasthana) [inline app.js]: We should move this to a constants file

File: app.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+app.use(bodyParser.urlencoded({ extended: true }))
+
+app.listen(port, () => {
+    console.log(`Listening to requests on http://localhost:${port}`);
+});
+
+client.connect(err => {
+  // perform actions on the collection object
+  if(!err) console.log("db connected");
+  const db = client.db('habit-app');
```
Comment (karanasthana) [inline app.js]: We should move this `habit-app` to a constants file

File: app.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const usersCollection = db.collection('users')
```
Comment (karanasthana) [inline app.js]: We should move this `users` to a constants file

File: app.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+app.use(bodyParser.urlencoded({ extended: true }))
+
+app.listen(port, () => {
+    console.log(`Listening to requests on http://localhost:${port}`);
+});
+
+client.connect(err => {
+  // perform actions on the collection object
+  if(!err) console.log("db connected");
```
Comment (karanasthana) [inline app.js]: Let's also make a helper function for console.log (which is only called in DEBUG mode, in future when we have these modes :P)

File: app.js (hunk 3 above)
Comment (ayu-ayushisood) [reply app.js]: We can't move collection name to a constant file, because at this moment we don't have a definite number of collection names to save them.

**Review by karanasthana — APPROVED** [review]
Very unclear about the structuring right now. Especially the node_modules.

**Pull request merged**